src/server/game/Handlers/QuestHandler.cpp

```diff
@@ -867,3 +867,31 @@ void WorldSession::HandlePlayerChoiceResponse(WorldPackets::Quest::ChoiceRespons
             _player->GetReputationMgr().ModifyReputation(sFactionStore.AssertEntry(faction.Id), faction.Quantity);
     }
 }
+
+void WorldSession::HandleUiMapQuestLinesRequest(WorldPackets::Quest::UiMapQuestLinesRequest& UiMap)
+{
+    UiMapEntry const* uiMap = sUiMapStore.LookupEntry(UiMap.UiMapID);
+    if (!uiMap)
+        return;
+
+    WorldPackets::Quest::UiMapQuestLinesResponse response;
+    response.UiMapID = uiMap->ID;
+
+    for (QuestPOIBlobEntry const* questPOIBlob : sQuestPOIBlobStore)
+        if (int32(uiMap->ID) == questPOIBlob->UiMapID)
+            if (_player->MeetPlayerCondition(questPOIBlob->PlayerConditionID))
+                for (QuestLineXQuestEntry const* questLineXQuest : sQuestLineXQuestStore)
+                    if (questPOIBlob->QuestID == questLineXQuest->QuestID)
+                        if (std::vector<QuestLineXQuestEntry const*> const* questLines = sDB2Manager.GetQuestsOrderForQuestLine(questLineXQuest->QuestLineID))
+                            for (QuestLineXQuestEntry const* questLineQuest : *questLines)
+                                if (Quest const* quest = sObjectMgr->GetQuestTemplate(questLineQuest->QuestID))
+                                    if (_player->CanTakeQuest(quest, false))
+                                        if(ContentTuningEntry const* contentTuning = sContentTuningStore.LookupEntry(quest->GetContentTuningId()))
+                                            if(_player->GetLevel() >= contentTuning->MinLevel)
+                                            {
+                                                response.QuestLineXQuestIDs.push_back(questLineQuest->ID);
+                                                break;
+                                            }
```

Comment:
Dude.. 😭🥲

Reply:
A more constructive comment: the nesting of if-conditions is too strong here. Either split these up and use returns or create a boolean variable with a lambda function which checks for all these points. So it's mostly the readability that suffers here.

Reply:
I find the explanation very good, but my knowledge of TrinityCore servers is not very advanced. Feel free to make any changes you deem appropriate.

Reply:
He meant you're making a cluster out of every if function which affects readability, this would help:

```suggestion
for (QuestPOIBlobEntry const* questPOIBlob : sQuestPOIBlobStore)
{
    if (int32(uiMap->ID) == questPOIBlob->UiMapID)
    {
        if (_player->MeetPlayerCondition(questPOIBlob->PlayerConditionID))
        {
            for (QuestLineXQuestEntry const* questLineXQuest : sQuestLineXQuestStore)
            {
                if (questPOIBlob->QuestID == questLineXQuest->QuestID)
                {
                    if (std::vector<QuestLineXQuestEntry const*> const* questLines = sDB2Manager.GetQuestsOrderForQuestLine(questLineXQuest->QuestLineID))
                    {
                        for (QuestLineXQuestEntry const* questLineQuest : *questLines)
                        {
                            if (Quest const* quest = sObjectMgr->GetQuestTemplate(questLineQuest->QuestID))
                            {
                                if (_player->CanTakeQuest(quest, false))
                                {
                                    if (ContentTuningEntry const* contentTuning = sContentTuningStore.LookupEntry(quest->GetContentTuningId()))
                                    {
                                        if(_player->GetLevel() >= contentTuning->MinLevel)
                                            response.QuestLineXQuestIDs.push_back(questLineQuest->ID);
                                            break;
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}
```

Reply:
Normally it's the person pulling requests doing the appropriate changes when reviewed, but declustering this many if-conditions would help readability because they're too many. This is just a suggestion, and in no way it's the better form to handle this many ifs cases; even @Ovahlord proposition is better, but:

```suggestion
for (QuestPOIBlobEntry const* questPOIBlob : sQuestPOIBlobStore)
{
    if (int32(uiMap->ID) == questPOIBlob->UiMapID)
    {
        if (_player->MeetPlayerCondition(questPOIBlob->PlayerConditionID))
        {
            for (QuestLineXQuestEntry const* questLineXQuest : sQuestLineXQuestStore)
            {
                if (questPOIBlob->QuestID == questLineXQuest->QuestID)
                {
                    if (std::vector<QuestLineXQuestEntry const*> const* questLines = sDB2Manager.GetQuestsOrderForQuestLine(questLineXQuest->QuestLineID))
                    {
                        for (QuestLineXQuestEntry const* questLineQuest : *questLines)
                        {
                            if (Quest const* quest = sObjectMgr->GetQuestTemplate(questLineQuest->QuestID))
                            {
                                if (_player->CanTakeQuest(quest, false))
                                {
                                    if (ContentTuningEntry const* contentTuning = sContentTuningStore.LookupEntry(quest->GetContentTuningId()))
                                    {
                                        if(_player->GetLevel() >= contentTuning->MinLevel)
                                            response.QuestLineXQuestIDs.push_back(questLineQuest->ID);
                                            break;
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}
```

Reply:
a better way would be inverting logic and then just using continues:

```c

for (QuestPOIBlobEntry const* questPOIBlob : sQuestPOIBlobStore)
{
    if (int32(uiMap->ID) != questPOIBlob->UiMapID)
        continue

    if (!_player->MeetPlayerCondition(questPOIBlob->PlayerConditionID))
        continue;

    for (QuestLineXQuestEntry const* questLineXQuest : sQuestLineXQuestStore)
    {
        if (questPOIBlob->QuestID != questLineXQuest->QuestID)
            continue;

        std::vector<QuestLineXQuestEntry const*> const* questLines = sDB2Manager.GetQuestsOrderForQuestLine(questLineXQuest->QuestLineID);
        if (!questLines)
            continue;

        for (QuestLineXQuestEntry const* questLineQuest : *questLines)
        {
            Quest const* quest = sObjectMgr->GetQuestTemplate(questLineQuest->QuestID);
            if (!quest)
                continue;
                
            if (!_player->CanTakeQuest(quest, false))
                continue;

            ContentTuningEntry const* contentTuning = sContentTuningStore.LookupEntry(quest->GetContentTuningId());
            if (!contentTuning)
                continue;

            if(_player->GetLevel() < contentTuning->MinLevel)
                continue;

            response.QuestLineXQuestIDs.push_back(questLineQuest->ID);
            break;
        }
    }
}
```

Reply:
This looks even better, use his recommendation!